### PR TITLE
Compilation fixes

### PR DESCRIPTION
**Pull request by winterheart**
Additional compilation fixes:

* Fixed pointer-to-int-cast warnings from compiler
* Fixed incomplete printf() format strings
* Replaced int to size_t in some places
* Fixed compilation for X11 and GTK apps

### CODE REVIEW

**Comment by rzumer** [comment]
Sorry for the delay, I will need some more time to review, as most of it looks good, but there are a few things I want to verify before merging.

File: src/menu.c [hunk 1]
```diff
@@ -137,9 +135,13 @@ static  void    start_file_error_dialog(int drv, int result)
     char wk[128];
     const t_menulabel *l = (drv<0) ? data_err_file : data_err_drive;
 
-    if (result == ERR_NO) return;
-    if (drv < 0) sprintf(wk, GET_LABEL(l, result));
-    else         sprintf(wk, GET_LABEL(l, result), drv+1);
+    if (result == ERR_NO)
+        return;
+    if (drv < 0)
+        // FIXME there should be 3rd number to fill printf-string
```
Comment (rzumer) [inline src/menu.c]: As seen on line 136, if `drv<0`, the label (defined in `message.h`) has no parameter to format. There is no warning or error generated with gcc [here](https://onlinegdb.com/B1kv8uTAB). Even if another compiler or configuration generates warnings this is not the right solution.

File: src/menu.c [hunk 2]
```diff
@@ -693,8 +695,10 @@ static  int get_dipsw_b2(void)
 }
 static  void    cb_dipsw_b2(UNUSED_WIDGET, void *p)
 {
-    if ((int)p) reset_req.boot_from_rom = TRUE;
-    else        reset_req.boot_from_rom = FALSE;
+    if ((intptr_t)p)
```
Comment (rzumer) [inline src/menu.c]: Please do not change the format of the code, such as adding line breaks, unless it is done consistently over the whole project, in a separate PR. I also prefer this format, but consistency is more important.

File: src/menu.c [hunk 3]
```diff
@@ -757,11 +761,13 @@ static  int get_dipsw_r(int p)
...
 static  void    cb_dipsw_r(UNUSED_WIDGET, void *p)
 {
-    int shift = data_dipsw_r[ (int)p >> 1 ].val;
-    int on    = (int)p & 1;
+    int shift = data_dipsw_r[ (intptr_t)p >> 1 ].val;
+    int on    = (intptr_t)p & 1;
 
-    if (on) reset_req.boot_dipsw |=  (1 << shift);
-    else    reset_req.boot_dipsw &= ~(1 << shift);
+    if (on)
```
Comment (rzumer) [inline src/menu.c]: As above.

File: src/menu.c [hunk 4]
```diff
@@ -1290,36 +1296,36 @@ static  void    cb_cpu_boost(Q8tkWidget *widget, void *mode)
...
+        }
     }
 
     if (fit) {              /* 適用した値が有効範囲なら、セット */
-    if (1 <= val && val <= 100) {
-        if (boost != val) {
-        boost_change(val);
+        if (1 <= val && val <= 100) {
+            if (boost != val) {
+            boost_change(val);
```
Comment (rzumer) [inline src/menu.c]: Correcting indentation is OK, but this inner block is also missing one tab.

File: src/menu.c [hunk 5]
```diff
@@ -1688,12 +1694,16 @@ static  int get_graph_pcg(void)
...
     if (widget) {
-    use_pcg = (int)p;
-    memory_set_font();
+        use_pcg = (intptr_t)p;
+        memory_set_font();
     }
 
-    if (use_pcg) { q8tk_widget_set_sensitive(graph_font_widget, FALSE); }
-    else         { q8tk_widget_set_sensitive(graph_font_widget, TRUE);  }
+    if (use_pcg) {
```
Comment (rzumer) [inline src/menu.c]: As requested above, please revert the format.

File: src/menu.c [hunk 6]
```diff
@@ -1814,15 +1824,31 @@ static  void    cb_volume(Q8tkWidget *widget, void *p)
...
-    switch ((int)p) {
-    case VOL_TOTAL:  xmame_cfg_set_mastervolume(vol);            break;
-    case VOL_FM:     xmame_cfg_set_mixer_volume(XMAME_MIXER_FM, vol);    break;
-    case VOL_PSG:    xmame_cfg_set_mixer_volume(XMAME_MIXER_PSG, vol);   break;
-    case VOL_BEEP:   xmame_cfg_set_mixer_volume(XMAME_MIXER_BEEP, vol);  break;
-    case VOL_RHYTHM: xmame_cfg_set_mixer_volume(XMAME_MIXER_RHYTHM, vol);break;
-    case VOL_ADPCM:  xmame_cfg_set_mixer_volume(XMAME_MIXER_ADPCM, vol); break;
-    case VOL_FMGEN:  xmame_cfg_set_mixer_volume(XMAME_MIXER_FMGEN, vol); break;
-    case VOL_SAMPLE: xmame_cfg_set_mixer_volume(XMAME_MIXER_SAMPLE, vol);break;
+    switch ((intptr_t)p) {
```
Comment (rzumer) [inline src/menu.c]: As above.

File: src/menu.c [hunk 7]
```diff
@@ -2651,11 +2677,13 @@ static  void    cb_disk_attr_protect_clicked(UNUSED_WIDGET, void *p)
 
     dialog_destroy();
 
-    if ((int)p) {
-    if ((int)p == 1) c = DISK_PROTECT_TRUE;
-    else             c = DISK_PROTECT_FALSE;
+    if ((intptr_t)p) {
```
Comment (rzumer) [inline src/menu.c]: As above.

File: src/menu.c [hunk 8]
```diff
@@ -2780,11 +2808,19 @@ static  void    cb_disk_attr_clicked(UNUSED_WIDGET, void *p)
 
     dialog_destroy();
 
-    switch ((int)p) {
-    case DATA_DISK_ATTR_RENAME:  sub_disk_attr_rename(name);    break;
-    case DATA_DISK_ATTR_PROTECT: sub_disk_attr_protect(name);   break;
-    case DATA_DISK_ATTR_FORMAT:  sub_disk_attr_format(name);    break;
-    case DATA_DISK_ATTR_BLANK:   sub_disk_attr_blank();     break;
+    switch ((intptr_t)p) {
```
Comment (rzumer) [inline src/menu.c]: As above.

File: src/menu.c [hunk 9]
```diff
@@ -3597,15 +3633,18 @@ static  int get_key_cfg(int type)
...
-    return FALSE;
 }
 static  void    cb_key_cfg(Q8tkWidget *widget, void *type)
 {
     int key = (Q8TK_TOGGLE_BUTTON(widget)->active) ? TRUE : FALSE;
 
-    switch ((int)type) {
-    case DATA_KEY_CFG_TENKEY:   tenkey_emu  = (key) ? TRUE : FALSE; break;
-    case DATA_KEY_CFG_NUMLOCK:  numlock_emu = (key) ? TRUE : FALSE; break;
+    switch ((intptr_t)type) {
```
Comment (rzumer) [inline src/menu.c]: As above.

File: src/menu.c [hunk 10]
```diff
@@ -3826,8 +3865,10 @@ static  int get_key_softkey(int code)
 }
 static  void    cb_key_softkey(Q8tkWidget *button, void *code)
 {
-    if (Q8TK_TOGGLE_BUTTON(button)->active) softkey_press  ((int)code);
-    else                                    softkey_release((int)code);
+    if (Q8TK_TOGGLE_BUTTON(button)->active)
```
Comment (rzumer) [inline src/menu.c]: As above.

File: src/z80-debug.c [hunk 11]
```diff
@@ -1374,7 +1374,7 @@ int z80_line_disasm( z80arch *z80, word pc )
   switch( Inst->type ){
   case OP_NOTHING:
     printf_head( z80, pc, 1 );
-    printf( Inst->str );
+    printf( "%s", Inst->str );
```
Comment (rzumer) [inline src/z80-debug.c]: `Inst->str` is `char*` type. So it is like calling `printf()` with no format argument, which compiles correctly with no warning for me like in my comment on the `sprintf()`. Unless there is a situation where it generates warnings or errors, it seems unnecessary.

**Comment by winterheart** [comment]
@rzumer done.

**Comment by rzumer** [comment]
Please also split the GTK2 support changes into another PR, so I can merge your compilation fixes before reviewing that.

File: src/z80-debug.c [hunk 12]
```diff
@@ -1407,7 +1407,7 @@ int z80_line_disasm( z80arch *z80, word pc )
     switch( Inst->type ){
     case OP_NOTHING:
       printf_head( z80, pc, 2 );
-      printf( Inst->str );
+      printf("%s", Inst->str );
```
Comment (rzumer) [inline src/z80-debug.c]: For the same reason mentioned in the last set of review comments, this is unnecessary: `Inst->str` is `char*`, and valid in `printf()` as the only parameter.

File: src/message.h [hunk 13]
```diff
@@ -2517,7 +2517,7 @@ static  Q8tkWidget *MAKE_KEY_COMBO(Q8tkWidget *box,
 
     q8tk_combo_set_text(combo, initstr);
     q8tk_misc_set_size(combo, 6, 0);
-    q8tk_signal_connect(combo, "activate", callback, (void*)(p->val));
+    q8tk_signal_connect(combo, "activate", callback, (void*)((intptr_t)p->val));
```
Comment (rzumer) [inline src/message.h]:
```suggestion
    q8tk_signal_connect(combo, "activate", callback, (void *)((intptr_t)p->val));
```
For consistent spacing.

File: src/fmgen/psg.cpp [hunk 14]
```diff
@@ -118,7 +118,7 @@ void PSG::MakeEnvelopTable()
         2,2, 2,0, 2,1, 2,3, 1,1, 1,3, 1,2, 1,0,
     };
     static uint8 table2[4] = {  0,  0, 31, 31 };
-    static uint8 table3[4] = {  0,  1, -1,  0 };
+    static uint8 table3[4] = {  0,  1, 255,  0 };
```
Comment (rzumer) [inline src/fmgen/psg.cpp]:
```suggestion
    static uint8 table3[4] = {  0,  1, 255, 0 };
```
For alignment with the previous line.

**Comment by winterheart** [comment]
@rzumer done

File: src/menu.c [hunk 15]
```diff
@@ -1688,12 +1686,12 @@ static  int get_graph_pcg(void)
...
     if (widget) {
-    use_pcg = (int)p;
-    memory_set_font();
+        use_pcg = (intptr_t)p;
+        memory_set_font();
     }
 
-    if (use_pcg) { q8tk_widget_set_sensitive(graph_font_widget, FALSE); }
-    else         { q8tk_widget_set_sensitive(graph_font_widget, TRUE);  }
+    if (use_pcg)  { q8tk_widget_set_sensitive(graph_font_widget, FALSE);  }
```
Comment (rzumer) [inline src/menu.c]:
```suggestion
    if (use_pcg) { q8tk_widget_set_sensitive(graph_font_widget, FALSE); }
```
Extra spaces don't seem necessary.

File: src/menu.c [hunk 16]
```diff
@@ -1688,12 +1686,12 @@ static  int get_graph_pcg(void)
...
+    else          { q8tk_widget_set_sensitive(graph_font_widget, TRUE);   }
```
Comment (rzumer) [inline src/menu.c]:
```suggestion
    else         { q8tk_widget_set_sensitive(graph_font_widget, TRUE);  }
```
As above.

File: src/menu.c [hunk 17]
```diff
@@ -1814,15 +1812,15 @@ static  void    cb_volume(Q8tkWidget *widget, void *p)
...
-    case VOL_TOTAL:  xmame_cfg_set_mastervolume(vol);            break;
-    case VOL_FM:     xmame_cfg_set_mixer_volume(XMAME_MIXER_FM, vol);    break;
-    case VOL_PSG:    xmame_cfg_set_mixer_volume(XMAME_MIXER_PSG, vol);   break;
-    case VOL_BEEP:   xmame_cfg_set_mixer_volume(XMAME_MIXER_BEEP, vol);  break;
-    case VOL_RHYTHM: xmame_cfg_set_mixer_volume(XMAME_MIXER_RHYTHM, vol);break;
-    case VOL_ADPCM:  xmame_cfg_set_mixer_volume(XMAME_MIXER_ADPCM, vol); break;
-    case VOL_FMGEN:  xmame_cfg_set_mixer_volume(XMAME_MIXER_FMGEN, vol); break;
-    case VOL_SAMPLE: xmame_cfg_set_mixer_volume(XMAME_MIXER_SAMPLE, vol);break;
+    switch ((intptr_t)p) {
+        case VOL_TOTAL:   xmame_cfg_set_mastervolume(vol);                        break;
```
Comment (rzumer) [inline src/menu.c]: This `break` statement is misaligned.

File: src/menu.c [hunk 18]
```diff
@@ -2780,11 +2776,11 @@ static  void    cb_disk_attr_clicked(UNUSED_WIDGET, void *p)
 
     dialog_destroy();
 
-    switch ((int)p) {
-    case DATA_DISK_ATTR_RENAME:  sub_disk_attr_rename(name);    break;
-    case DATA_DISK_ATTR_PROTECT: sub_disk_attr_protect(name);   break;
-    case DATA_DISK_ATTR_FORMAT:  sub_disk_attr_format(name);    break;
-    case DATA_DISK_ATTR_BLANK:   sub_disk_attr_blank();     break;
+    switch ((intptr_t)p) {
+    case DATA_DISK_ATTR_RENAME:   sub_disk_attr_rename(name);   break;
```
Comment (rzumer) [inline src/menu.c]:
```suggestion
    case DATA_DISK_ATTR_RENAME:  sub_disk_attr_rename(name);   break;
```
Extra space is unnecessary.

File: src/menu.c [hunk 19]
```diff
@@ -2780,11 +2776,11 @@ static  void    cb_disk_attr_clicked(UNUSED_WIDGET, void *p)
...
+    case DATA_DISK_ATTR_PROTECT:  sub_disk_attr_protect(name);  break;
```
Comment (rzumer) [inline src/menu.c]:
```suggestion
    case DATA_DISK_ATTR_PROTECT: sub_disk_attr_protect(name);  break;
```
As above.

File: src/menu.c [hunk 20]
```diff
@@ -2780,11 +2776,11 @@ static  void    cb_disk_attr_clicked(UNUSED_WIDGET, void *p)
...
+    case DATA_DISK_ATTR_FORMAT:   sub_disk_attr_format(name);   break;
```
Comment (rzumer) [inline src/menu.c]:
```suggestion
    case DATA_DISK_ATTR_FORMAT:  sub_disk_attr_format(name);   break;
```
As above.

File: src/menu.c [hunk 21]
```diff
@@ -2780,11 +2776,11 @@ static  void    cb_disk_attr_clicked(UNUSED_WIDGET, void *p)
...
+    case DATA_DISK_ATTR_BLANK:    sub_disk_attr_blank();        break;
```
Comment (rzumer) [inline src/menu.c]:
```suggestion
    case DATA_DISK_ATTR_BLANK:   sub_disk_attr_blank();        break;
```
As above.

**Comment by winterheart** [comment]
@rzumer done

File: src/menu.c [hunk 22]
```diff
@@ -1814,15 +1812,15 @@ static  void    cb_volume(Q8tkWidget *widget, void *p)
...
-    case VOL_TOTAL:  xmame_cfg_set_mastervolume(vol);            break;
-    case VOL_FM:     xmame_cfg_set_mixer_volume(XMAME_MIXER_FM, vol);    break;
-    case VOL_PSG:    xmame_cfg_set_mixer_volume(XMAME_MIXER_PSG, vol);   break;
-    case VOL_BEEP:   xmame_cfg_set_mixer_volume(XMAME_MIXER_BEEP, vol);  break;
-    case VOL_RHYTHM: xmame_cfg_set_mixer_volume(XMAME_MIXER_RHYTHM, vol);break;
-    case VOL_ADPCM:  xmame_cfg_set_mixer_volume(XMAME_MIXER_ADPCM, vol); break;
-    case VOL_FMGEN:  xmame_cfg_set_mixer_volume(XMAME_MIXER_FMGEN, vol); break;
-    case VOL_SAMPLE: xmame_cfg_set_mixer_volume(XMAME_MIXER_SAMPLE, vol);break;
+    switch ((intptr_t)p) {
+        case VOL_TOTAL:   xmame_cfg_set_mastervolume(vol);                    break;
```
Comment (rzumer) [inline src/menu.c]: There are some extra spaces in this `switch` too after colons.

File: src/menu.c [hunk 23]
```diff
@@ -1688,12 +1686,12 @@ static  int get_graph_pcg(void)
...
     if (widget) {
-    use_pcg = (int)p;
-    memory_set_font();
+        use_pcg = (intptr_t)p;
+        memory_set_font();
     }
 
-    if (use_pcg) { q8tk_widget_set_sensitive(graph_font_widget, FALSE); }
-    else         { q8tk_widget_set_sensitive(graph_font_widget, TRUE);  }
+    if (use_pcg) { q8tk_widget_set_sensitive(graph_font_widget, FALSE);  }
```
Comment (rzumer) [inline src/menu.c]: Extra spaces before closing braces on the next 2 lines.

File: src/menu.c [hunk 24]
```diff
@@ -693,8 +691,8 @@ static  int get_dipsw_b2(void)
 }
 static  void    cb_dipsw_b2(UNUSED_WIDGET, void *p)
 {
-    if ((int)p) reset_req.boot_from_rom = TRUE;
-    else        reset_req.boot_from_rom = FALSE;
+    if ((intptr_t)p)  reset_req.boot_from_rom = TRUE;
```
Comment (rzumer) [inline src/menu.c]:
```suggestion
    if ((intptr_t)p) reset_req.boot_from_rom = TRUE;
```
An extra space.

File: src/menu.c [hunk 25]
```diff
@@ -693,8 +691,8 @@ static  int get_dipsw_b2(void)
...
+    else              reset_req.boot_from_rom = FALSE;
```
Comment (rzumer) [inline src/menu.c]:
```suggestion
    else             reset_req.boot_from_rom = FALSE;
```
For realignment with the previous line.

**Comment by winterheart** [comment]
@rzumer done

File: src/X11/graph-x11dga.c [hunk 26]
```diff
@@ -7,11 +7,7 @@
 #ifdef  USE_DGA
 #include <unistd.h>
 #include <sys/types.h>
-#if 1   /* obsolete? */
```
Comment (rzumer) [inline src/X11/graph-x11dga.c]: Why was the default include removed instead of the unused one?

Comment (winterheart) [reply src/X11/graph-x11dga.c]: Because use of <X11/extensions/xf86dga.h> is deprecated more than 8 years ago.

Comment (rzumer) [reply src/X11/graph-x11dga.c]: OK.

**Pull request merged**